tests/test_transforms.py

```diff
@@ -647,6 +647,189 @@ def test_integer_shift_photon():
+    """Test several ways to flip a profile
+    """
+    import time
+    t1 = time.time()
+
+    # The Shapelet profile has the advantage of being fast and not circularly symmetric, so
+    # it is a good test of the actual code for doing the flips (in SBTransform).
+    # But since the bug Rachel reported in #645 was actually in SBInterpolatedImage
+    # (one calculation implicitly assumed dx > 0), it seems worthwhile to run through all the
+    # classes to make sure we hit everything with negative steps for dx and dy.
+    prof_list = [
+        galsim.Shapelet(sigma=0.17, order=2,
+                        bvec=[1.7, 0.01,0.03, 0.29, 0.33, -0.18]),
+    ]
+    if __name__ == "__main__":
+        image_dir = './real_comparison_images'
+        catalog_file = os.path.join(image_dir,'test_catalog.fits')
+        rgc = galsim.RealGalaxyCatalog(catalog_file, image_dir)
+        # Some of these are slow, so only do the Shapelet test as part of the normal unit tests.
+        prof_list += [
+            galsim.Airy(lam_over_diam=0.17, flux=1.7),
+            galsim.Airy(lam_over_diam=0.17, obscuration=0.2, flux=1.7),
+            # Box gets rendered with real-space convolution.  The default accuracy isn't quite
+            # enough to get the flip to match at 6 decimal places.
+            galsim.Box(0.17, 0.23, flux=1.7,
+                       gsparams=galsim.GSParams(realspace_relerr=1.e-6)),
+            # Without being convolved by anything with a reasonable k cutoff, this needs
+            # a very large fft.
+            galsim.DeVaucouleurs(half_light_radius=0.17, flux=1.7,
+                                 gsparams=galsim.GSParams(maximum_fft_size=8000)),
+            # I don't really understand why this needs a lower maxk_threshold to work, but
+            # without it, the k-space tests fail.
+            galsim.Exponential(scale_radius=0.17, flux=1.7,
+                               gsparams=galsim.GSParams(maxk_threshold=1.e-4)),
+            galsim.Gaussian(sigma=0.17, flux=1.7),
+            galsim.Kolmogorov(fwhm=0.17, flux=1.7),
+            galsim.Moffat(beta=2.5, fwhm=0.17, flux=1.7),
+            galsim.Moffat(beta=2.5, fwhm=0.17, flux=1.7, trunc=0.82),
+            galsim.OpticalPSF(lam_over_diam=0.17, obscuration=0.2, nstruts=6,
+                              coma1=0.2, coma2=0.5, defocus=-0.1, flux=1.7),
+            # Like with Box, we need to increase the real-space convolution accuracy.
+            # This time lowering both relerr and abserr.
+            galsim.Pixel(0.23, flux=1.7,
+                         gsparams=galsim.GSParams(realspace_relerr=1.e-6,
+                                                  realspace_abserr=1.e-8)),
+            # Note: RealGalaxy should not be rendered directly because of the deconvolution.
+            # Here we convolve it by a Gaussian that is slightly larger than the original PSF.
+            galsim.Convolve([ galsim.RealGalaxy(rgc, index=0, flux=1.7),  # "Real" RealGalaxy
```

Comment:
Why are these "Real" and "Fake"?  They look like they are just two entries from the catalog, both convolved with a Gaussian as necessary to render an image.